src/debugging/Inspector/scripts/generate.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python
+
+import json
+from codegen import *
+
+_PRIMITIVE_TO_TS_NAME_MAP = {
+    'integer': 'number',
+    'object': 'Object'
+}
+
+_ALWAYS_UPPERCASED_ENUM_VALUE_SUBSTRINGS = set(['API', 'CSS', 'DOM', 'HTML', 'XHR', 'XML'])
+
+def load_specification(protocol, filepath, isSupplemental=False):
+    try:
+        with open(filepath, "r") as input_file:
+            parsed_json = json.load(input_file)
+            protocol.parse_specification(parsed_json, isSupplemental)
+    except ValueError as e:
+        raise Exception("Error parsing valid JSON in file: " + filepath)
+
+def generate(combined_domains_path, output_dir):
+    protocol = models.Protocol("JavaScriptCore")
+    load_specification(protocol, combined_domains_path);
+    protocol.resolve_types()
+
+    generator = TypeScriptInterfaceGenerator(protocol)
+    output = generator.generate_output()
+
+    output_file = open(os.path.join(output_dir, generator.output_filename()), "w")
+    output_file.write(output)
+    output_file.close()
+
+def stylized_name_for_enum_value(enum_value):
```

Comment:
What's the difference between this method and the base `stylized_name_for_enum_value`?